CRM/Contribute/Page/Tab.php

```diff
@@ -131,44 +130,52 @@ public function browse() {
     $controller->run();
 
     // add recurring block
-    $action = array_sum(array_keys($this->recurLinks()));
-    $params = CRM_Contribute_BAO_ContributionRecur::getRecurContributions($this->_contactId);
+    try {
```

Comment:
@omarabuhussein Yes that's a good idea.  It would be nice to refactor this in future to either create a separate tab, or introduce sortable columns etc. on the existing tab.  Moving to the API means this is supported by the code, moving it to it's own function makes any future refactoring easier.